### DISCUSSION TITLE
SendEmailError on SMTP with SSL : Connection unexpectedly closed

### DISCUSSION THREAD

**Discussion by galbini** [fief-dev/fief#402] [Q&A]
## Describe the bug

During registration of a new user, the following exception occurred:

`  
File "/usr/local/lib/python3.12/site-packages/fief/tasks/register.py", line 33, in run
    self.email_provider.send_email(
    │    │              └ <function SMTP.send_email at 0xffff79e47600>
    │    └ <fief.services.email.smtp.SMTP object at 0xffff785e1dc0>
    └ <fief.tasks.register.OnAfterRegisterTask object at 0xffff784750a0>
  File "/usr/local/lib/python3.12/site-packages/fief/services/email/smtp.py", line 60, in send_email
    raise SendEmailError(str(e)) from e
          └ <class 'fief.services.email.base.SendEmailError'>
`

## To Reproduce

1. Configure Fief with SMTP with a SSL connexion  ex : EMAIL_PROVIDER_PARAMS={"ssl": true, "host": "mail.infomaniak.com", "port":"465"}
2. Go to '/register'
3. Set form and Click on 'subcribe'
4. See error on logs

## Configuration

* Cloud or self-hosted: self-hosted Getting start mode
* If self-hosted, Fief version: latest

## Additional context

My SMTP config work with https://github.com/turbodog/python-smtp-mail-sending-tester
In your code, the SSL parameter seems used to configure tls but not ssl with SMTP_SSL object.

**Comment by fief-bailiff** [comment]
Hail, @galbini 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to [star our repository](https://github.com/fief-dev/fief) ⭐️

<p align="center">
<img width="656" alt="star-fief" src="https://github.com/fief-dev/fief/assets/1144727/b2db7f59-102f-47e5-9751-67fa10372c52">
</p>

## Want to support us?

Subscribe to one of our paid plan to help us continue our work and receive exclusive information and benefits! Starts at $5/month 🪙

<p align="center">
    <a href="https://polar.sh/fief-dev">
    <picture>
      <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/embed/subscribe.svg?org=fief-dev&darkmode=1">
      <img alt="Subscribe" src="https://polar.sh/embed/subscribe.svg?org=fief-dev">
    </picture>
    </a>
</p>

Farewell!

**Comment by frankie567** [comment]
Yes, our SMTP implementation doesn't support SSL encryption but rather TLS encryption. This is something that has been [reported a few times](https://github.com/orgs/fief-dev/discussions/234#discussioncomment-6374491) (not sure if/how we should fix this).

Try with the port `587` instead of `465`, that should fix the issue.

**Comment by galbini** [comment]
Hello,
I try with port 587 but I was the next error
fief.services.email.base.SendEmailError: (550, b'5.7.1 Sender mismatch')

I'm not a python expert but i found the next sample code that's works

`
server = None
if options.usessl:
	server = smtplib.SMTP_SSL(serveraddr, options.serverport)
else:
	server = smtplib.SMTP(serveraddr, options.serverport)

server.set_debuglevel(options.debuglevel)
server.connect(serveraddr, options.serverport)
server.ehlo()
if options.usetls: server.starttls()
server.ehlo()
if options.SMTP_USER != "": server.login(options.SMTP_USER, options.SMTP_PASS)
server.sendmail(fromaddr, toaddr, msg)
server.quit()
`

**Comment by frankie567** [comment]
If I understand the SMTP error `5.7.1 Sender mismatch`, it means that the SMTP server rejected the `From` address you specified. Make sure to set the `DEFAULT_FROM_EMAIL` environment variable to an address authorized by your provider.

**Comment by galbini** [comment]
Thanks, that's work !
